ui/lib/config-ui/addon/components/messages/page/list.hbs

```diff
@@ -115,4 +115,34 @@
       data-test-action-text="Create message"
     />
   </EmptyState>
+{{/if}}
+
+{{#if this.showMaxMessageModal}}
+  <Hds::Modal
+    id="maximum-message-modal"
+    @size="large"
+    @color="warning"
+    @onClose={{fn (mut this.showMaxMessageModal) false}}
+    data-test-modal="maximum-message-modal"
+    as |M|
+  >
+    <M.Header data-test-modal-title="maximum-message-modal">
+      Maximum number of messages reached
+    </M.Header>
+    <M.Body data-test-modal-body="maximum-message-modal">
+      Vault can only store up to 100 messages. To create a message, delete one of your messages to clear up space.
+    </M.Body>
+    <M.Footer>
+      <Hds::Button
+        @text="Confirm"
+        {{on "click" (fn (mut this.showMaxMessageModal) false)}}
+        data-test-modal-button="maximum-message-modal"
+      />
+      <Hds::Button
```

Comment:
Since both buttons do the same thing I don't think we need both. Elsewhere in the app we just render a single `Close` button for these cases
<img width="836" alt="Screenshot 2024-01-19 at 12 47 33 PM" src="https://github.com/hashicorp/vault/assets/68122737/22f95858-1a00-477d-8de2-287ad23dcd2a">